### PR TITLE
[release-4.5] Bug 1880318: Update k8s version to v0.18.6

### PR DESCRIPTION
**Pull request by tremes**
Bugfix that consisted of K8s/OpenShift version updates.

## Categories

- [X] Bugfix
- [ ] Enhancement
- [ ] Backporting
- [ ] Others (CI, Infrastructure, Documentation)

## Sample archive

No new data.

## Documentation

- no need to update

## Unit Tests

- no tests needed for this bugfix

## Privacy

 no new data

## References

https://bugzilla.redhat.com/show_bug.cgi?id=1880318

### CODE REVIEW

File: pkg/config/configobserver/configobserver_test.go [hunk 1]
```diff
@@ -75,9 +75,9 @@ func TestChangeSupportConfig(t *testing.T) {
 				defaultConfig: ctrl,
 			}
 			c.mergeConfigLocked()
-			err := c.retrieveToken()
+			err := c.retrieveToken(context.Background())
 			if err == nil {
-				err = c.retrieveConfig()
+				err = c.retrieveConfig(context.Background())
```
Comment (rluders) [inline pkg/config/configobserver/configobserver_test.go:80]: Also, do we need 2 contexts?

File: pkg/config/configobserver/configobserver.go [hunk 2]
```diff
@@ -41,10 +41,10 @@ func New(defaultConfig config.Controller, kubeClient kubernetes.Interface) *Cont
 		checkPeriod:   5 * time.Minute,
 	}
 	c.mergeConfigLocked()
-	if err := c.retrieveToken(); err != nil {
+	if err := c.retrieveToken(context.TODO()); err != nil {
 		klog.Warningf("Unable to retrieve initial token config: %v", err)
 	}
-	if err := c.retrieveConfig(); err != nil {
+	if err := c.retrieveConfig(context.TODO()); err != nil {
```
[the remark this answers is not in the document]
Comment (tremes) [reply pkg/config/configobserver/configobserver.go:47]: I did it the same as in master - see https://github.com/openshift/insights-operator/blob/master/pkg/config/configobserver/configobserver.go#L44. `TODO()` returns an empty context. Maybe we could pass the existing context, but there's no need to do so IMO and I would start this change in master.

File: pkg/config/configobserver/configobserver_test.go (hunk 1 above)
Comment (tremes) [reply pkg/config/configobserver/configobserver_test.go:80]: Basically same answer as above. In case of test it's not so important IMO.

Comment (rluders) [reply pkg/config/configobserver/configobserver_test.go:80]: Agreed.

File: pkg/config/configobserver/configobserver.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (rluders) [reply pkg/config/configobserver/configobserver.go:47]: Yeh, I know. I was asking just by curiosity. At the first it looked a little bit overhead to create 2 context, but ... I guess that it is fine.

**Review by rluders — APPROVED** [review]

File: pkg/config/configobserver/configobserver.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (tremes) [reply pkg/config/configobserver/configobserver.go:47]: Sure np. It was good question. Thanks.

**Pull request merged**